### PR TITLE
✨ GitLab: Security Policy check

### PR DESCRIPTION
**Pull request by raghavkaul**
#### What kind of change does this PR introduce?
Adds GitLab support for the Security-Policy check. Also changes `Repo.Org()` to align better with its named meaning.

#### What is the new behavior (if this is a feature change)?**
Checks for Security policy file in the repo. Doesn't attempt to check for Org-level security policy (no equivalent to `.github` folder in GitLab). `GetOrganizationSecurityPolicy` is exposed as a method of RepoClient.

- [x] Tests for the changes have been added (for bug fixes/features)

#### Does this PR introduce a user-facing change?

```release-note
GitLab support for Security-Policy check
```

### CODE REVIEW

File: checks/raw/security_policy_test.go [hunk 1]
```diff
@@ -138,7 +138,7 @@ func TestSecurityPolicy(t *testing.T) {
 			mockRepo := mockrepo.NewMockRepo(ctrl)
 
 			mockRepoClient.EXPECT().ListFiles(gomock.Any()).Return(tt.files, nil).AnyTimes()
-			mockRepo.EXPECT().Org().Return(nil).AnyTimes()
+			mockRepo.EXPECT().Org().Return("").AnyTimes()
```
Comment (azeemshaikh38) [inline checks/raw/security_policy_test.go:141]: Curious why this change was needed?

**Review by azeemshaikh38 — APPROVED** [review]

**Review by azeemshaikh38 — APPROVED** [review]
Nice PR, thanks. LGTM with comments.

File: checks/raw/security_policy_test.go (hunk 1 above)
Comment (raghavkaul) [reply checks/raw/security_policy_test.go]: This was initially because the return type of `Org()` was changed from RepoClient to string (the latter is not nilable)

File: clients/repo_client.go [hunk 2]
```diff
@@ -40,6 +41,7 @@ type RepoClient interface {
 	GetCreatedAt() (time.Time, error)
 	GetDefaultBranchName() (string, error)
 	GetDefaultBranch() (*BranchRef, error)
+	GetOrgPolicyRepoClient(context.Context) (RepoClient, error)
```
[the remark this answers is not in the document]
Comment (raghavkaul) [reply clients/repo_client.go:44]: There's no Org repo, `.github` is where the Org's community policy is stored.

[the remark this answers is not in the document]
Comment (azeemshaikh38) [reply clients/repo_client.go:44]: Not really true that `.github` is used only for org community policy? AFAIK `.github` is generally used to store any org-level PR templates, workflows etc. See https://github.com/sigstore/.github for example.

**Comment by azeemshaikh38** [comment]
> I'd argue changing an interface is a breaking change.

That's fair. Maybe remove all occurences of `Org()` in `checks` package, mark the API as deprecated in `RepoClient` and we can remove the APi before v5 release. Would that work?

**Comment by spencerschrock** [comment]
> > I'd argue changing an interface is a breaking change.
> 
> That's fair. Maybe remove all occurences of `Org()` in `checks` package, mark the API as deprecated in `RepoClient` and we can remove the APi before v5 release. Would that work?

that works for Org in the Repo interface. But RepoClient had a method added which is also a breaking change.

**Comment by spencerschrock** [comment]
in terms of breaking changes, if the only breakage is Allstar (who has a mock client that would need to stub the new method), it's probably safe to do if we give Allstar a heads up. And it's a small enough that we could send the update PR

**Comment by azeemshaikh38** [comment]
> in terms of breaking changes, if the only breakage is Allstar (who has a mock client that would need to stub the new method), it's probably safe to do if we give Allstar a heads up. And it's a small enough that we could send the update PR

Can Allstar use Scorecard's mock client - https://github.com/ossf/scorecard/blob/main/clients/mockclients/repo_client.go ?

**Comment by raghavkaul** [comment]
Thanks Spencer for pointing that out, I wasn't previously aware that people relied on Scorecard's interfaces. I gave a heads up to the Allstar maintainer and they said that they are aware of breakages and will upgrade manually.
